simulator/gdb/sim_wrapper.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * sim_wrapper.cpp - Functional simulator wrappers for GDB
+ * @author Vyacheslav Kompan
+ * Copyright 2018 MIPT-MIPS
+ */
+
+#include "sim_wrapper.h"
+
+/* MIPT-MIPS simulator interfaces */
+#include <simulator.h>
+#include <infra/config/config.h>
+#include <mips/mips.h>
+#include <func_sim/func_sim.h>
+
+#define ISA_NAME "mips32"
+
+typedef struct {
+    std::unique_ptr<Simulator> ptr = nullptr;
+    std::string filename;
+} SimulatorInstance;
+static SimulatorInstance simInstance;
+
+
+/* Create simulator */
+void mips_sim_create (const char **argv, const char *filename) try {
+    /* Get argc
+     * Passed arguments start at argv[2], end with NULL */
+    int argc = 0;
+    while (argv[2 + argc++]);
+    /* However, POPL seems to skips first passed argument, so we pass starting at
+     * argv[1], which is "--sysroot=" */
+    config::handleArgs (argc, &argv[1]);
+
+    /* Using dynamic_cast here would be a huge overhead */
+    //TODO: add simulator arguments
+    simInstance.ptr = Simulator::create_simulator (ISA_NAME, true, true);
```

Comment:
Why not just pass "mips32" explicitly instead of macro? I don't see it used anywhere else, whereas the function creates only mips simulation.

Reply:
Just to make it easier to change architecture if needed

Reply:
If we add new architecture (e.g. RISCV), we would introduce new wrapping functions anyway.